cmd/infra/powervs/create.go

```diff
@@ -0,0 +1,1192 @@
+	err = checkPowerVSRegion(options.PowerVSRegion)
+	if err != nil {
+		return
+	}
+
+	infra := &Infra{ID: options.InfraID}
+
+	defer func() {
+		out := os.Stdout
+		if len(options.OutputFile) > 0 {
+			var err error
+			out, err = os.Create(options.OutputFile)
+			if err != nil {
+				log(options.InfraID).Error(err, "cannot create output file")
+			}
+			defer out.Close()
+		}
+		outputBytes, err := json.MarshalIndent(infra, "", "  ")
+		if err != nil {
+			log(options.InfraID).WithName(options.InfraID).Error(err, "failed to serialize output infra")
+		}
+		_, err = out.Write(outputBytes)
+		if err != nil {
+			log(options.InfraID).Error(err, "failed to write output infra json")
+		}
+	}()
+
+	err = infra.SetupInfra(options)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+// SetupInfra ...
+// infra creation orchestration
+func (infra *Infra) SetupInfra(options *CreateInfraOptions) (err error) {
+	startTime := time.Now()
+
+	log(options.InfraID).Info("Setup infra started")
+
+	// if IBMCLOUD_API_KEY is not set, infra cannot be set up.
+	if cloudApiKey == "" {
+		return fmt.Errorf("IBMCLOUD_API_KEY not set")
+	}
+
+	infra.ResourceGroupID, err = getResourceGroupID(options.ResourceGroup)
+	if err != nil {
+		return fmt.Errorf("error getting id for resource group %s, %w", options.ResourceGroup, err)
+	}
+
+	err = infra.setupBaseDomain(options)
+	if err != nil {
+		return fmt.Errorf("error setup base domain: %w", err)
+	}
+
+	v1, err := createVpcService(options.VpcRegion, options.InfraID)
+	if err != nil {
+		return fmt.Errorf("error creating vpc service: %w", err)
+	}
+
+	err = infra.setupVpc(options, v1)
+	if err != nil {
+		return fmt.Errorf("error setup vpc: %w", err)
+	}
+
+	err = infra.setupVpcSubnet(options, v1)
+	if err != nil {
+		return fmt.Errorf("error setup vpc subnet: %w", err)
+	}
+
+	session, err := createPowerVSSession(options.PowerVSRegion, options.PowerVSZone, options.Debug)
+	infra.AccountID = session.Options.UserAccount
+	if err != nil {
+		return fmt.Errorf("error creating powervs session: %w", err)
+	}
+
+	err = infra.setupPowerVSCloudInstance(options)
+	if err != nil {
+		return fmt.Errorf("error setup powervs cloud instance: %w", err)
+	}
+
+	err = infra.setupPowerVSCloudConnection(options, session)
+	if err != nil {
+		return fmt.Errorf("error setup powervs cloud connection: %w", err)
+	}
+
+	err = infra.setupPowerVSDhcp(options, session)
+	if err != nil {
+		return fmt.Errorf("error setup powervs dhcp server: %w", err)
+	}
+
+	err = infra.isCloudConnectionReady(options, session)
+	if err != nil {
+		return fmt.Errorf("cloud connection is not up: %w", err)
+	}
+
+	log(options.InfraID).Info("Setup infra completed in", "duration", time.Since(startTime).String())
+	return
+}
+
+// getIAMAuth...
+// getting core.Authenticator
+func getIAMAuth() *core.IamAuthenticator {
+	return &core.IamAuthenticator{
+		ApiKey: cloudApiKey,
+	}
+}
+
+// setupBaseDomain ...
+// get domain id and crn of given base domain
+// To-Do
```

Comment:
can we please stick to "TODO(username): action" convention?

Reply:
Modified the comment to this convention